allennlp/semparse/contexts/table_question_context.py

```diff
@@ -0,0 +1,130 @@
+        	column_name_sempre = curr_line[2]
+        	column_idx = curr_line[1]
+        	column_name = column_name_sempre[11:]
+        	column_idx_to_name[column_idx] = column_name
+            idx += 1
+
+        
+        column_node_type_info = [{'string' : 0, 'number' : 0, 'date' : 0} 
+                                 for col in column_idx_to_name]
+        cell_values = set()
+        while idx < len(lines):
+        	curr_line = lines[idx]
+            column_idx = curr_line[1]
+            node_info = zip(header, curr_line)
+            cell_values.add(cls._normalize_string(node_info['content']))  
+            if node_info['date']:
+            	column_node_type_info[column_idx]['date'] += 1 
+            # If cell contains too many tokens, then likely not number
+            elif node_info['number'] and num_tokens < max_tokens_for_num_cell:
+            	column_node_type_info[column_idx]['number'] += 1 
+            else:
+            	column_node_type_info[column_idx]['string'] += 1
+
+
+        return cls(cell_values, column_node_type_info, column_idx_to_name)
+
+
+    def get_entities_from_question(self, question : List[str]):
+        entity_data = []
+        for i, token in enumerate(question):
+            if token in STOP_WORDS: continue
+            normalized_token = self._normalize_string(token)
+            if len(normalized_token) == 0: continue
+            if self._string_in_table(normalized_token):
+                curr_data = {'value' : normalized_token, 'token_start' : i, 'token_end' : i+1}
+                entity_data.append(curr_data)
+                                  
+
+        expanded_entities = self._expand_entities(question, entity_data, table)
+        return expanded_entities #TODO(shikhar) Handle conjunctions
+    
+        
+
+    def _string_in_table(self, candidate : str) -> bool:
+        for cell_value in self.cell_values:
+            if candidate in cell_value: return True
+        return False
+
+
+    def _process_conjunction(self, entity_data):
+        raise NotImplementedError
+
+    def _expand_entities(self, question, entity_data):
+        new_ents = []
+        for ent in entity_data:
+            # to ensure the same strings are not used over and over
+            if new_ents and ent['token_end'] <= new_ents[-1]['token_end']: continue
+            curr_st = ent['token_start']
+            curr_en = ent['token_end']
+            curr_tok = ent['value']
+
+            while curr_en < len(question): 
+                next_tok = question[curr_en]
+                next_tok_normalized = self._normalize_string(next_tok) 
+                candidate = "%s_%s" %(curr_tok, next_tok_normalized) 
+                if self._string_in_table(candidate):
+                    curr_en += 1
+                    curr_tok = candidate
+                else: 
+                    break
+
+            new_ents.append({'token_start' : curr_st, 'token_end' : curr_en 'value' : curr_tok})
```

Comment:
Missing comma between `curr_en` and `value`

Reply:
Done, this seems like an older file, i didn't commit/push the one with these syntax issues fixed. :/ So, these have been taken care of.